infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogFiles.java

```diff
@@ -725,7 +742,23 @@ public void append(LogAddress logAddress, LogData entry) {
             SegmentHandle fh = getSegmentHandleForAddress(logAddress);
             if (fh.getKnownAddresses().containsKey(logAddress.address) ||
                     fh.getTrimmedAddresses().contains(logAddress.address)) {
-                throw new OverwriteException();
+                if (entry.getRank()==null) {
+                    throw new OverwriteException();
+                } else {
+                    try (MultiReadWriteLock.AutoCloseableLock ignored = segmentLocks.acquireWriteLock(fh.getSegment())) {
```

Comment:
Since append is serially called, is this lock necessary? also not write locks are acquired in write record too.